server/checks/PullRequest.js

```diff
@@ -0,0 +1,99 @@
+import Check from './Check'
+import { logger } from '../../common/debug'
+import * as EVENTS from '../model/GithubEvents'
+
+const CHECK_TYPE = 'pullrequest'
+const ACTIONS = ['opened', 'edited', 'reopened', 'synchronize']
+
+const ISSUE_PATTERN = /^[^#\s]*#\d+$/
+// Grubber's pattern
+const URL_PATTERN = /\b((?:[a-z][\w-]+:(?:\/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}\/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’]))/i
+
+const info = logger(CHECK_TYPE, 'info')
+const error = logger(CHECK_TYPE, 'error')
+
+const status = (description, state = 'success') => ({
+  description,
+  state,
+  context: 'zappr/check/pull-request'
+})
+
+const isLongEnough = (str, requiredLength) => (str || '').length > requiredLength
+const containsPattern = pattern => str => (str || '').split(' ')
+  .findIndex(s => pattern.test(s)) > -1
+const containsUrl = containsPattern(URL_PATTERN)
+const containsIssueNumber = containsPattern(ISSUE_PATTERN)
+
+export default class PullRequest extends Check {
+  static TYPE = CHECK_TYPE
+  static NAME = 'Pull request check'
+  static HOOK_EVENTS = [EVENTS.PULL_REQUEST]
+
+  /**
+   * @param {GithubService} github
+   */
+  constructor(github) {
+    super() // TODO: create constructor that receives `github`
```

Comment:
Ah, okay. (Because what the `TODO` points out is already done.)

Well, I'm not sure. The only other check to have an own status function is `approval`, and it's quite complex. Probably not worth spending the effort now.